src/modules/royalty.ts

```diff
@@ -0,0 +1,152 @@
+import { Royalty, Royalty__factory } from "@3rdweb/contracts";
+import { BigNumberish } from "ethers";
+import { ModuleType, Role, RolesMap } from "../common";
+import { Currency, getCurrencyMetadata } from "../common/currency";
+import { Module } from "../core/module";
+
+interface IRoyalty {
```

Comment:
@jakeloo @jnsdls please let me know what you guys think about this interface before I fill out the methods.

Reply:
is this public interface?

Reply:
I'll make it public